crypto/crypto_aesctr_aesni.c

```diff
@@ -45,6 +46,9 @@ crypto_aesctr_aesni_stream_wholeblocks(struct crypto_aesctr * stream,
 	memcpy(block_counter_be_arr, stream->pblk + 8, 8);
 	block_counter = stream->bytectr / 16;
 
+	/* clang doesn't realize that we'll do the for() loop at least once. */
+	block_counter_be_sse = _mm_loadu_si64(stream->pblk + 8);
```

Comment:
In theory, we could remove this line and set `__m128i block_counter_be_sse = {0};` instead.  But if we somehow got to this function and didn't run the `for` loop, we'd end up with garbage in `pblk`.  I can't measure any speed difference between the versions, and I figured that this was slightly more clear.

Reply:
What happens if you remove this line and add `if (num_blocks > 0)` before the `_mm_storeu_si64`?

Reply:
Same warning -- clang is really worried about this variable!

Reply:
I added a REBASE commit in case you wanted to take a look at it.